### PR TITLE
improvement: allow adding extra dependencies to Ruby SDK

### PR DESCRIPTION
**Pull request by armandobelardo**

### CODE REVIEW

File: generators/ruby/sdk/src/CustomConfig.ts [hunk 1]
```diff
@@ -1,15 +1,17 @@
-import { BaseGeneratorConfigSchema } from "@fern-api/generator-commons";
+import { BaseGeneratorConfigSchema, ExtraDependenciesSchema } from "@fern-api/generator-commons";
 import { z } from "zod";
 
 export type RubySdkCustomConfig = z.infer<typeof RubySdkCustomConfigSchema>;
 export const RubySdkCustomConfigSchema = BaseGeneratorConfigSchema.extend({
     defaultTimeoutInSeconds: z.optional(z.union([z.literal("infinity"), z.number()])),
+    extraDevDependencies: z.optional(ExtraDependenciesSchema),
```
Comment (dsinghvi) [inline generators/ruby/sdk/src/CustomConfig.ts:7]: should we do `extraDependencies` as well while we are here?

File: generators/ruby/sdk/CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## Unreleased
 
+## [0.8.0-rc0] - 2024-07-01
+
+- Improvement: allow users to specify additional dependencies and dev dependencies for Ruby SDKs.
```
Comment (dsinghvi) [inline generators/ruby/sdk/CHANGELOG.md:12]: would be nice to add an example

**Review by dsinghvi — APPROVED** [review]

File: generators/ruby/sdk/src/CustomConfig.ts (hunk 1 above)
Comment (armandobelardo) [reply generators/ruby/sdk/src/CustomConfig.ts:7]: It's in the base object, so it's already there!

File: generators/ruby/sdk/CHANGELOG.md (hunk 2 above)
Comment (armandobelardo) [reply generators/ruby/sdk/CHANGELOG.md:12]: added!

**Pull request merged**